packages/app-store/routing-forms/zod.ts

```diff
@@ -55,6 +55,64 @@ const queryValueSchema = z.object({
   properties: z.any(),
 });
 
+/**
+ * Stricter schema for validating before saving to DB
+ */
+export const queryValueSaveValidationSchema = queryValueSchema
+  .omit({ children1: true })
+  .merge(
+    z.object({
+      children1: z
+        .record(
+          z.object({
+            type: z.string().optional(),
+            properties: z
+              .object({
+                field: z.any().optional(),
+                operator: z.any().optional(),
+                value: z.any().optional(),
+              })
+              .optional(),
+          })
+        )
+        .optional()
+        // Be very careful and lenient here. Just ensure that the rule isn't invalid without breaking anything
+        .superRefine((children1, ctx) => {
+          if (!children1) return;
+          const isObject = (value: unknown): value is Record<string, unknown> =>
+            typeof value === "object" && value !== null;
+          Object.entries(children1).forEach(([, _rule]) => {
+            const rule = _rule as unknown;
+            if (!isObject(rule) || rule.type !== "rule") return;
+            if (!isObject(rule.properties)) return;
+
+            const value = rule.properties.value || [];
+            if (!(value instanceof Array)) {
+              return;
+            }
```

Comment:
Be lenient with what is passed by RAQB and just focus on one thing which is the value.